syntax = "proto3";

package chat;

service ChatService {
    rpc SendMessage(ChatMessage) returns (ChatResponse);
    rpc StreamMessages(StreamRequest) returns (stream ChatMessage);
}

message ChatMessage {
    string username = 1;
    string message = 2;
    string timestamp = 3;
}

message ChatResponse {
    bool success = 1;
    string message = 2;
}

message StreamRequest {
    string username = 1;
}